src/runtime_src/core/tools/xbflash2/OO_Program_Qspips.cpp

```diff
@@ -0,0 +1,178 @@
+ * at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "OO_Program_Qspips.h"
+#include "tools/common/XBUtilitiesCore.h"
+
+#include "core/pcie/tools/xbflash.qspi/firmware_image.h"
+#include "core/pcie/tools/xbflash.qspi/pcidev.h"
+#include "core/pcie/tools/xbflash.qspi/xqspips.h"
+
+namespace XBU = XBUtilities;
+
+// 3rd Party Library - Include Files
+#include <boost/algorithm/string.hpp>
+#include <boost/format.hpp>
+#include <boost/program_options.hpp>
+
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <iostream>
+#include <fstream>
+
+namespace {
+
+void
+qspipsCommand(po::variables_map& vm)
+{
+    //root privileges required.
+    XBU::sudo_or_throw_err();
+
+    //mandatory command line args
+    std::string bdf = vm.count("device") ? vm["device"].as<std::string>() : "";
+    if (bdf.empty()) {
+        throw std::runtime_error("Device not specified. Please specify a single device using --device option");
+    }
+    //optional command line args
+    std::string flash_type = vm.count("flash-part") ? vm["flash-part"].as<std::string>() : "";
+    std::string soffset = vm.count("offset") ? vm["offset"].as<std::string>() : "";
+    size_t offset = !soffset.empty() ? std::stoul(soffset) : 0;
+    bool force = vm.count("force") ? true : false;
+    std::string sBar = vm.count("bar") ? vm["bar"].as<std::string>() : "";
+    int bar = !sBar.empty() ? std::stoi(sBar) : 0;
+    std::string sBarOffset = vm.count("bar-offset") ? vm["bar-offset"].as<std::string>() : "";
+    size_t baroff = !sBarOffset.empty() ? std::stoul(sBarOffset) : INVALID_OFFSET;   
+
+    if (vm.count("erase")) { //qspips erase
+        std::string slength = vm.count("length") ? vm["length"].as<std::string>() : "";
+        size_t len = !slength.empty()  ? std::stoi(slength) : GOLDEN_BASE;    
+
+        std::cout << "About to erase flash"
+            << boost::format(" [0x%x, 0x%x] on device %s\n") % offset % (offset + len) % bdf;
+
+        if (offset + len > GOLDEN_BASE)
+            std::cout << "\nProvided length from the offset is crossing golden image boundary, This might erase golden image.\n" << std::endl;
+
+        if (!force && !XBU::can_proceed())
+            return;
+
+        pcidev::pci_device dev(bdf, bar, baroff, flash_type);
+        XQSPIPS_Flasher qspips(&dev);
+
+        if (qspips.xclErase(offset, len))
+            throw std::runtime_error("qspips erase failed.");
+    }
+    else if (vm.count("image")) {// qspips flash 
```

Comment:
We shouldn't be using these nested ladders.  Instead, the previous if block should simply return successful.
```suggestion
    if (vm.count("image")) {// qspips flash 
```